### PR TITLE
Upgrade electron builder

### PR DESCRIPTION
**Pull request by posix4e**
- This allows us to use the prebuilt electron to build electron
- It still requires a rsync of muon into this directory
- Use a madeup muon version to chill electron
- Add appimage capable build
- Note this is electron's muon not muon muon
- level is required at runtime
- Support 32 or 64 bit builds
- RPM & DEB support directly off the prebuilt

TODO
- [ ] Add icons
- [ ] Setup homepage to work right, currently blank shows up
- [ ] Set mac category

Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Test Plan:

Reviewer Checklist:

Tests

- [x] Adequate test coverage exists to prevent regressions
- [x] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [x] New files have MPL2 license header

### CODE REVIEW

**Comment by posix4e** [comment]
Does anyone know what mac category to use?

**Comment by posix4e** [comment]
I was also able to build debs, rpms and packs this way. Perhaps we just have yarn run linux and it builds all of them?

**Comment by bbondy** [comment]
Awesome thanks @posix4e 

Yes I've been wanting to use electron-builder for this. I think it can also do snaps btw which would be good if you can add too.  FYI @w0ts0n to not do this work if it is being done here.

Please also adjust the `tools/buildInstaller.js` file (`npm run build-installer`)

**Comment by posix4e** [comment]
@bbondy true about the snaps!

**Comment by Liunkae** [comment]
Safari and Firefox both use `public.app-category.productivity` for the app category.

**Comment by luixxiul** [comment]
@posix4e would you mind opening an issue to track the progress? thanks in advance.

**Comment by posix4e** [comment]
@luixxiul how is https://github.com/brave/browser-laptop/issues/7857

**Comment by bsclifton** [comment]
Marking as work in progress since there are 3 open items:
https://github.com/brave/browser-laptop/issues/7857#issuecomment-327883475

File: package.json [hunk 1]
```diff
@@ -3,6 +3,28 @@
   "version": "0.21.0",
   "description": "Brave laptop and desktop browser",
   "main": "./app/index.js",
+  "build": {
+    "appId": "brave.com",
+    "electronDist": "node_modules/electron-prebuilt/dist",
+    "mac": {
+      "category": "public.app-category.productivity"
+    },
+    "muonVersion": "1.2.3.4",
```
Comment (NejcZdovc) [inline package.json]: Just a reminder that we need to update this value, before merging

**Comment by cezaraugusto** [comment]
@posix4e sorry closing this as stale for now please re-open when you have a window. Thanks!

**Comment by posix4e** [comment]
Sounds good

**Pull request closed without merging**